### ISSUE TITLE
Update Analytics module datapoints for consistency

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#450]
Datapoints in the Analytics module should receive minor updates as outlined in the ACs.

Implementation brief should include which pieces of JS need to be updated and which e2e tests need to be updated (if any).

**Execution is blocked until #406 is merged.**

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Analytics setup and querying Analytics data via API should still work as before.
* Parameters and return values controlled by the plugin itself should all be in camel case.
* Datapoints should be updated as follows:
    * GET connection → keep as is
    * GET account-id → keep as is
    * GET property-id → keep as is
    * GET profile-id → keep as is
    * GET internal-web-property-id → keep as is
    * GET use-snippet → keep as is
    * GET amp-client-id-opt-in → keep as is
    * GET goals → keep as is
    * GET get-accounts → GET accounts-properties-profiles
        * odd name but used because it returns all that; we wanna keep ourselves the chance for adding a real GET accounts in the future
    * GET get-properties → GET properties-profiles
        * see above; at some point we might wanna improve the client codebase and actually have a proper GET properties
    * GET get-profiles → GET profiles
    * GET tag-permission → keep as is
    * GET adsense & GET site-analytics & GET top-pages & GET overview & GET traffic-sources → GET report
        * additional supported (optional) parameters should be “dimensions” (list of strings), “metrics” (list of objects with “expression” and optional “alias” property), and “orderby” (list of objects with “fieldName” and “sortOrder” property)
        * “pageUrl” parameter should be updated to use “permaLink” like the others
parsed response should only return a single object from “reports” since it will always be just that single one
        * if metrics contain any expression starting with “ga:adsense”, the parse_data_response() method should set/delete the option whether Analytics and AdSense are linked
    * POST connection → keep as is
    * POST account-id → keep as is
    * POST property-id → keep as is
    * POST profile-id → keep as is
    * POST internal-web-property-id → keep as is
    * POST use-snippet → keep as is
    * POST amp-client-id-opt-in → keep as is
    * POST save → POST settings

#### `GET get-accounts`
* Rename datapoint to `accounts-properties-profiles`
* [update usage](https://github.com/google/site-kit-wp/search?q=get-accounts&unscoped_q=get-accounts)

#### `GET get-properties`
* Rename datapoint to `properties-profiles`
* [update usage](https://github.com/google/site-kit-wp/search?q=get-properties&unscoped_q=get-properties)

#### `GET get-profiles`
* Rename datapoint to `profiles`
* [update usage](https://github.com/google/site-kit-wp/search?q=get-profiles&unscoped_q=get-profiles)

#### `GET adsense + site-analytics + top-pages + overview + traffic-sources`
* Combine into a single new datapoint `GET report`
* Add support for additional optional parameters:
    - `dimensions` (list of strings)
    - `metrics` (list of objects with `expression` and optional `alias` properties)
    - `orderby` (list of objects with `fieldName` and `sortOrder` properties)
* Update usage of `adsense` datapoint
    - AdSenseDashboardWidgetTopPagesTableSmall
    - AnalyticsAdSenseDashboardWidgetTopPagesTable
* Update usage of `site-analytics` datapoint
    - AnalyticsDashboardWidgetSiteStats
    - AnalyticsDashboardWidgetTopLevel
* Update usage of `top-pages` datapoint
    - AnalyticsDashboardWidgetPopularPagesTable
    - AnalyticsDashboardWidgetTopPagesTable
    - WPAnalyticsDashboardWidgetTopPagesTable
* Update usage of `overview` datapoint
    - AnalyticsAdminbarWidgetOverview
    - AnalyticsDashboardWidgetOverview
    - AnalyticsDashboardWidgetTopLevel
    - WPAnalyticsDashboardWidgetOverview
    - publisher-wins/index.js `googlesitekit.PublisherWinsNotification` (multiple)
* Update usage of `traffic-sources` datapoint
    - AnalyticsAllTrafficDashboardWidgetTopAcquisitionSources
    - AnalyticsDashboardWidgetTopAcquisitionSources
    - DashboardAcquisitionPieChart
    - isDataZeroForTrafficSources
* Remove old datapoints: `adsense`, `site-analytics`, `top-pages`, `overview`, `traffic-sources`

* Parameter has already been renamed in #406
> “pageUrl” parameter should be updated to use “permaLink” like the others  

#### VRT

These changes will likely break VRT so we will need to review and approve. Test data may also need to be regenerated.

## Changelog entry

* Improve datapoint behavior of Analytics module for consistency.

**Comment by aaemnnosttv** [comment]
@felixarntz for functions like `isDataZeroForTrafficSources`, should this be renamed/repurposed for Report?

**Comment by felixarntz** [comment]
@aaemnnosttv 
> for functions like `isDataZeroForTrafficSources`, should this be renamed/repurposed for Report?

Note that there is also already a `isDataZeroForReporting()` which seems to do almost exactly the same, except that is more comprehensive (for whatever reason `isDataZeroForTrafficSources()` only looks at the first three array entries.

Preferably the two would be consolidated to `isDataZeroForReporting()`, combining the best of both functions (the first one has parts where the code is a bit clunky / hard to follow, the second is more limited in functionality).

IB generally looks great ✅

**Comment by felixarntz** [comment]
This will be blocked until #500 is done. Once unblocked, move back to QA column.
